junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -1993,6 +1993,24 @@ public static Set<Class<?>> getAllAssignmentCompatibleClasses(Class<?> clazz) {
 		return result;
 	}
 
+	/**
+	 * {@return whether the supplied {@code object} is an instance of a record class}
+	 * @since 1.12
+	 */
+	@API(status = INTERNAL, since = "1.12")
+	public static boolean isRecordObject(Object object) {
```

Comment:
Let's move these new methods to directly after `isInnerClass()`,

Reply:
Done in 6c020d50294c98fdee506eb768f5765761c9052e